packages/core/src/fhir-deduplication/resources/medication-administration.ts

```diff
@@ -0,0 +1,53 @@
+import { MedicationAdministration } from "@medplum/fhirtypes";
+import { combineResources, fillMaps, getDateFromResource } from "../shared";
+
+export function deduplicateMedAdmins(medications: MedicationAdministration[]): {
+  combinedMedAdmins: MedicationAdministration[];
+  refReplacementMap: Map<string, string[]>;
+} {
+  const { medAdminsMap, remainingMedAdmins, refReplacementMap } = groupSameMedAdmins(medications);
+  return {
+    combinedMedAdmins: combineResources({
+      combinedMaps: [medAdminsMap],
+      remainingResources: remainingMedAdmins,
+    }),
+    refReplacementMap,
+  };
+}
+
+/**
+ * Approach:
+ * 1 map, where the key is made of:
+ * - medicationReference ID
+ * - status
+ * - date
+ */
+export function groupSameMedAdmins(medAdmins: MedicationAdministration[]): {
+  medAdminsMap: Map<string, MedicationAdministration>;
+  remainingMedAdmins: MedicationAdministration[];
+  refReplacementMap: Map<string, string[]>;
+} {
+  const medAdminsMap = new Map<string, MedicationAdministration>();
+  const refReplacementMap = new Map<string, string[]>();
+  const remainingMedAdmins: MedicationAdministration[] = [];
+
+  for (const medAdmin of medAdmins) {
+    const medRef = medAdmin.medicationReference?.reference;
+    const date = getDateFromResource(medAdmin, "date-hm");
+    const status = medAdmin.status;
+    const dosage = medAdmin.dosage;
+
+    if (medRef) {
+      const key = JSON.stringify({ medRef, status, date, dosage });
```

Comment:
ditto here re status as a merge condition and not a deduplication condition. Any reason you used dosage here but didnt use it in MedicationStatement?

Reply:
Yea, so the reason was that here it's an object, but in MedicationStatement, it's an array, so we can combine those. Altho, thinking about it, I'm not sure that's good cuz having 2 different dosage entries for the same deduplicated resource might not work very well. But I think we should look at the data together before we decide.